### PR TITLE
propose RFC: transports

### PR DESCRIPTION
**Pull request by TelegramSam**
Signed-off-by: Sam Curren <telegramsam@gmail.com>

### CODE REVIEW

File: features/0025-didcomms-transports/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## Reference
+
+Standardized transport methods are detailed here. 
+
+### HTTP(S)
+
+HTTP(S) is the first transport for DID Communication that has received heavy attention.
+
+- Messages are transported via HTTP POST.
+- The MIME Type for the POST request is `application/ssi-agent-wire`.
```
Comment (troyronda) [inline features/0025-didcomms-transports/README.md]: Is this mime type remaining as "-wire"?

File: features/0025-didcomms-transports/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- Using Secure Websockets (wss://) with TLS 1.2 or greater with a forward secret cipher will provide Perfect Forward Secrecy (PFS) on the transmission leg.
+
+#### Known Implementations
+
+[Python Reference Agent](<https://github.com/hyperledger/indy-agent/tree/master/python>)
+
+
+### Other Transports
+
+Other transports may be used for Agent messaging. As they are developed, this HIPE should be updated with appropriate standards for the transport method. A PR should be raised against this doc to facilitate discussion of the proposed additions and/or updates. New transports should highlight the common elements of the transport (such as an HTTP response code for the HTTP transport) and how they should be applied.
```
Comment (troyronda) [inline features/0025-didcomms-transports/README.md]: s/HIPE/RFC/

File: features/0025-didcomms-transports/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### Message Routing
+
+The transports described here are used between two agents. In the case of [message routing](<https://github.com/hyperledger/indy-hipe/tree/master/text/0022-cross-domain-messaging>), a message will travel across multiple agent connections. Each intermediate agent (see [Mediators and Relays](<https://github.com/hyperledger/indy-hipe/tree/master/text/0036-mediators-and-relays>)) may use a different transport. These transport details are not made known to the sender, who only knows the keys of Mediators and the first endpoint of the route.
```
Comment (troyronda) [inline features/0025-didcomms-transports/README.md]: Are these referenced HIPES being moved to RFCs?

Comment (dhh1128) [reply features/0025-didcomms-transports/README.md]: I will be moving the one about mediators and relays.

File: features/0025-didcomms-transports/README.md (hunk 1 above)
Comment (dhh1128) [reply features/0025-didcomms-transports/README.md]: I suggest that we rename this to `application/didcomm-enc-envelope` or similar.

Comment (TelegramSam) [reply features/0025-didcomms-transports/README.md]: I reversed envelope-enc, but this is great.

File: features/0025-didcomms-transports/README.md (hunk 2 above)
Comment (TelegramSam) [reply features/0025-didcomms-transports/README.md]: fixed. Thanks!

**Comment by kdenhartog** [comment]
Did you want to keep working on this until merged or just merge new stuff on later? I know we talked about splitting this into a transport concept RFC and then 2 feature RFCs. I'm fine merging them and then making changes as needed. Does anyone else have a differing opinion?

**Comment by TelegramSam** [comment]
@kdenhartog Let's do the merge, and update as we add the split out RFCs

**Pull request merged**